rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/PropertiesManagement.scala

```diff
@@ -469,6 +470,45 @@ class PropertiesManagement extends DispatchSnippet with Loggable {
     ) apply (xml ++ Script(Run("correctButtons();") & check()))
   }
 
+  def networkProtocolSection = { xml : NodeSeq =>
+    //  initial values, updated on successfull submit
+    var initReportsProtocol = configService.report_protocol
+    var reportProtocol = initReportsProtocol.getOrElse("tcp")
+    def check() = {
+      S.notice("updateNetworkProtocol","")
+      Run(s"""$$("#networkProtocolSubmit").button( "option", "disabled",${initReportsProtocol.map(_ == reportProtocol).getOrElse(false)});""")
+    }
+
+    def submit = {
+      configService.set_report_protocol(reportProtocol).foreach(updateOk => initReportsProtocol = Full(reportProtocol))
+
+      // start a promise generation, Since we check if there is change to save, if we got there it mean that we need to redeploy
+      startNewPolicyGeneration
+      S.notice("updateNetworkProtocol","Network protocol options correctly updated")
+      check()
+    }
+
+     ( "#reportProtocol" #> {
+      initReportsProtocol match {
+        case Full(value) =>
+          SHtml.ajaxCheckbox(
+              value == "udp"
+            , (b : Boolean) => { reportProtocol = if (b) "udp" else "tcp"; check() }
```

Comment:
Could you please build an algebraic data structure for that ?
trait SyslogProtocol { def name: String }
case object SyslogUDAP { val name = "udp" } extends SyslogProtocol
case object SyslogTCP { val name = "tcp" }  extends SyslogProtocol

So that we don't have typo and refactoring problems in the future ?